tools/run-dev

```diff
@@ -204,135 +198,46 @@ def start_webpack_watcher() -> "subprocess.Popen[bytes]":
-    newpath = urlunsplit((protocol, host, path, query, ""))
-    return newpath
-
-
-client: httpclient.AsyncHTTPClient
-
-
-class BaseHandler(web.RequestHandler):
-    # target server ip
-    target_host: str = "127.0.0.1"
-    # target server port
-    target_port: int
-
-    def _add_request_headers(
-        self,
-        exclude_lower_headers_list: Sequence[str] = [],
-    ) -> httputil.HTTPHeaders:
-        headers = httputil.HTTPHeaders()
-        for header, v in self.request.headers.get_all():
-            if header.lower() not in exclude_lower_headers_list:
-                headers.add(header, v)
-        return headers
-
-    @override
-    def get(self) -> None:
-        pass
-
-    @override
-    def head(self) -> None:
-        pass
-
-    @override
-    def post(self) -> None:
-        pass
-
-    @override
-    def put(self) -> None:
-        pass
-
-    @override
-    def patch(self) -> None:
-        pass
-
-    @override
-    def options(self) -> None:
-        pass
-
-    @override
-    def delete(self) -> None:
-        pass
-
-    @override
-    async def prepare(self) -> None:
-        assert self.request.method is not None
-        assert self.request.remote_ip is not None
-        if "X-REAL-IP" not in self.request.headers:
-            self.request.headers["X-REAL-IP"] = self.request.remote_ip
-        if "X-FORWARDED_PORT" not in self.request.headers:
-            self.request.headers["X-FORWARDED-PORT"] = str(proxy_port)
-        url = transform_url(
-            self.request.protocol,
-            self.request.path,
-            self.request.query,
-            self.target_port,
-            self.target_host,
+session: aiohttp.ClientSession
+
+
+async def forward(upstream_port: int, request: web.Request) -> None:
+    async with session.request(
+        request.method,
+        request.url.with_host("127.0.0.1").with_port(upstream_port),
+        headers=[
+            (key, value) for key, value in request.headers.items() if key != hdrs.TRANSFER_ENCODING
+        ],
+        data=None if request.method in ("GET", "HEAD", "OPTIONS") else request.content.iter_any(),
+        allow_redirects=False,
+        auto_decompress=False,
+        compress=False,
+        skip_auto_headers={hdrs.ACCEPT, hdrs.ACCEPT_ENCODING, hdrs.CONTENT_TYPE, hdrs.USER_AGENT},
+    ) as upstream_response:
+        response = web.StreamResponse(
+            status=upstream_response.status, reason=upstream_response.reason
         )
-        try:
-            request = httpclient.HTTPRequest(
-                url=url,
-                method=self.request.method,
-                headers=self._add_request_headers(["upgrade-insecure-requests"]),
-                follow_redirects=False,
-                body=self.request.body,
-                allow_nonstandard_methods=True,
-                # use large timeouts to handle polling requests
-                connect_timeout=240.0,
-                request_timeout=240.0,
```

Comment:
Does the new proxy have any timeout before it kills requests? I don't see a clear reason it would need one from a practical standpoint, but seems worth knowing.

Reply:
aiohttp uses a default total timeout of 5 minutes, which seemed fine.

https://docs.aiohttp.org/en/stable/client_quickstart.html#timeouts